mt-kahypar/partition/refinement/fm/strategies/gain_cache_on_demand_strategy.h

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
```

Comment:
usually we have the convention to summarize such strategies in a folder with name "policies"